lib/connectors/memory.js

```diff
@@ -207,17 +207,24 @@ Memory.prototype.create = function create(model, data, options, callback) {
   var idName = this.idName(model);
   id = (props[idName] && props[idName].type && props[idName].type(id)) || id;
   this.setIdValue(model, data, id);
-  if(!this.collection(model)) {
+  if (!this.collection(model)) {
     this.collection(model, {});
   }
 
   if (this.collection(model)[id]) {
-    return process.nextTick(function() {
-      callback(new Error('Duplicate entry for ' + model + '.' + idName));
-    });
+      throw new Error('Duplicate entry for ' + model + '.' + idName);
+      return;
   }
-
   this.collection(model)[id] = serialize(data);
+  return id;
+}
+
+Memory.prototype.create = function create(model, data, options, callback) {
+  try {
+    var id = this._create(model, data);
+  } catch (e) {
+    return callback(e);
+  }
```

Comment:
Mixing callback-based error handling with try/catch is not a good idea. Since most Node.js code never throws, it's not a common practice to wrap calls in try/catch blocks, thus we may forget to handle errors in future changes and cause crashes.

There is also (micro)optimization concern: IIRC, V8 does not optimize functions containing try/catch blocks.

I am proposing to rework the `_create` method to accept the usual error-first callback function.


```js
Memory.prototype.create = function(...) {
  var self = this;
  this._create(model, data, function(err, id) {
    if (err) return callback(err);
    // ...
  });
}
```